### DISCUSSION TITLE
Turbsim windfield size for FAST.Farm

### DISCUSSION THREAD

**Discussion by MYMahfouz** [OpenFAST/openfast#1562] [Q&A]
I am trying to generate a windfield for FAST.Farm using Turbsim.

My farm width is 3000m and grid height is 500m. I am using the 15MW IEA wind turbine so I am using a resolution of 5m to be less than the max chord length. The dt I am using is 0.25s.

When I try to produce this huge windfield in Turbsim it fails with the error Invalid size.

Is there a way to get over this problem?
How much can I increase my resolution and dt without affecting FAST.Farm results?

Thanks in advance,
BR,
Youssef

**Comment by jjonkman** [comment]
Dear @MYMahfouz,

This is why the FAST.Farm modeling guidance (https://openfast.readthedocs.io/en/dev/source/user/fast.farm/ModelGuidance.html) generally recommends that when using TurbSim to generate synthetic inflow for FAST.Farm, that you should use `Mod_AmbWind` = 3 so that you can use separate low- (large but coarse) and high-resolution (small but fine) domains.

If you are not already aware, there have been quite a few improvements recently to the python scrips that support FAST.Farm, particularly the inflow generation setup with TurbSim, see: https://github.com/OpenFAST/python-toolbox.

Best regards,

**Comment by MYMahfouz** [comment]
Dear @jjonkman,
Thanks for your reply. I was not aware of the possibility of using a high and low wind fields for the low and high resolution domains. I need to read more about this option.

I am aware of pyFAST and I was using it for wind field generation in TurbSim, but to the best of my understanding there is no example at the moment for generating two separate wind fields with different resolutions. 
I am working on similar functions to use Mann turbulence generator, and will share them when I am done.

BR,

**Reply by jjonkman** [reply]
Dear @MYMahfouz,

I would suggest looking at the following pull request for the python toolbox, where many improvements to the python scripts have been developed: https://github.com/OpenFAST/python-toolbox/pull/56.

I'm not aware that a similar functionality could be developed for the Mann model, because the TurbSim approach makes use of the constrained turbulence option of TurbSim (known as `TIMESR` in TurbSim), and I'm not aware of a constrained turbulence option for the Mann model.

Best regards,

**Reply by MYMahfouz** [reply]
Dear @jjonkman,
Thanks yes this pull request should do what I am trying to do. Thanks I will check it out.

I am sorry I didn't explain myself properly. I didn't mean to use the Mann Box to create coarse and high resolution field. I just meant that I am working on similar functions to create the FAST Farm model and Mann Box together from the use inputs. Similar to example one and two currently in the main branch of pyFAST.

BR,
